<a href="https://colab.research.google.com/github/Rafa11-ai/data-science-2024/blob/main/Pertemuan2_Rafa_Hidayah_240401010203.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Nama  : Rafa Hidayah

NIM   : 240401010203

Kelas : IF405

In [ ]:
import pandas as pd
import numpy as np

# Muat Dataset Titanic
URL = "https://raw.githubusercontent.com/datasciencedojo/datasets/master/titanic.csv"
df = pd.read_csv(URL)

# Eksplorasi Awal
print(df.shape, df.columns.tolist())
print(df.isnull().sum())

# Filtering - Penumpang Wanita Kelas 1
wk1 = df[(df['Sex']=='female') &
 (df['Pclass']==1)]
print(wk1['Survived'].mean(), wk1['Age'].mean())

# Groupby - Tingkat Keselamatan per Kelas
hasil = df.groupby('Pclass').agg(
    total=('PassengerId','count'),
    selamat=('Survived','mean')
).round(3)
print(hasil)

# Presentase penumpang yang selamat
selamat = np.sum(df['Survived']) / len(df) * 100
print(f"\nSurvived = {selamat:2f}%")

# Jumlah penumpang kelas 1
jumlah_wk1 = len(wk1)
print(f"Penumpang Wanita Kelas 1 = {jumlah_wk1}")

Survival_by_sex = df.groupby("Sex")["Survived"].mean() * 100

print("\nTingkat Keselamatan Berdasarkan Jenis Kelamin:")

print(f"Female : {Survival_by_sex['female']:.2f}%")
print(f"Male   : {Survival_by_sex['male']:.2f}%")

(891, 12) ['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp', 'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked']
PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64
0.9680851063829787 34.61176470588235
        total  selamat
Pclass                
1         216    0.630
2         184    0.473
3         491    0.242

Survived = 38.383838%
Penumpang Wanita Kelas 1 = 94

Tingkat Keselamatan Berdasarkan Jenis Kelamin:
Female : 74.20%
Male   : 18.89%


Q1.

12 Kolom terdiri dari:
1. PassengerId
2. Survived
3. Pclass
4. Name
5. Sex
6. Age
7. SibSp
8. Parch
9. Ticket
10. Fare
11. Cabin
12. Embarked

Di output program terdapat tampilan baris (891) dan kolom (12), serta detail jumlah data yang terisi di masing-masing kolom.

Q2.

Kolom Age : 177

Cabin     : 687

Embarked  : 2

Presentase = jumlah missing/891(jumlah baris data) x 100

Age = 177/891 x 100 = 19.87%

Cabin = 687/891 x 100 = 77.10%

Embarked = 2/891 x100 = 0.22%

Q3.
Presentase penumpang selamat = 38.383838%.
Saya menggunakan operasi Numpy.

Q4.
Jumlah Wanita Kelas 1 = 94.

Rata-rata usia = 34.61 tahun.

Q5.

1. Kelas (Pclass) 1:

- Jumlah Penumpang = 216

- Tingkat Keselamatan = 0.630


2. Kelas (Pclass) 2:

- Jumlah Penumpang = 184

- Tingkat Keselamatan = 0.473

3. Kelas (Pclass) 3:

- Jumlah Penumpang = 491

- Tingkat Keselamatan = 0.242

Q6.
Saya membuat analisis keselamatan berdasarkan jenis kelamin.

Tingkat Keselamatan Berdasarkan Jenis Kelamin:

Female : 74.20%

Male   : 18.89%

Interpretasi:

Penumpang perempuan memiliki peluang selamat lebih tinggi dibandingkan dengan laki-laki.

Apa yang dipelajari
- Memuat dan mengeksplorasi dataset menggunakan Pandas dan NumPy.
- Mengidentifikasi jumlah kolom, ukuran dataset, dan missing values.
- Melakukan filtering data berdasarkan kondisi tertentu.
- Menghitung statistik sederhana seperti rata-rata dan persentase.
- Membandingkan tingkat keselamatan berdasarkan kelas penumpang dan jenis kelamin.

Temuan utama
- Dataset Titanic berisi 891 baris dan 12 kolom.
- Missing value terbesar terdapat pada kolom Cabin (77,10%), diikuti Age (19,87%).
- Tingkat keselamatan keseluruhan sekitar 38,38%.
- Penumpang kelas 1 memiliki peluang selamat tertinggi (63%), sedangkan kelas 3 terendah (24,2%).
- Penumpang perempuan memiliki tingkat keselamatan jauh lebih tinggi (74,2%) dibandingkan laki-laki (18,89%).
- Terdapat 94 penumpang perempuan kelas 1 dengan rata-rata usia 34,61 tahun.

Keterbatasan / Pertanyaan yang muncul
- Analisis masih bersifat deskriptif dan belum mencari penyebab hubungan antarvariabel.
- Mengapa perempuan memiliki tingkat keselamatan yang jauh lebih tinggi dibandingkan laki-laki?
- Apakah usia juga memengaruhi peluang keselamatan penumpang?
- Bagaimana hasil analisis berubah jika missing value pada kolom Age diatasi terlebih dahulu?
- Apakah kombinasi kelas penumpang dan jenis kelamin memberikan pola keselamatan yang lebih jelas?
- Variabel mana yang paling berpengaruh terhadap keselamatan penumpang?